File: src/main/java/models/Galeries.java
package models;

public class Galeries {
    private static int nextId = 1; // Variable statique pour stocker le prochain identifiant à utiliser
    private int id;
    private String nom;
    private String adresse;
    private String ville;
    private String pays;
    private int capaciteMax;

    public Galeries(int id, String nom, String adresse, String ville, String pays, int capaciteMax) {
        this.id = id;
        this.nom = nom;
        this.adresse = adresse;
        this.ville = ville;
        this.pays = pays;
        this.capaciteMax = capaciteMax;
    }

    public Galeries() {
        this.id = generateId(); // Utilisez la méthode generateId pour obtenir un nouvel identifiant unique
    }

    public Galeries(String nom, String adresse, String ville, String pays, int capaciteMax) {

        this.nom = nom;
        this.adresse = adresse;
        this.ville = ville;
        this.pays = pays;
        this.capaciteMax = capaciteMax;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public String getPays() {
        return pays;
    }

    public void setPays(String pays) {
        this.pays = pays;
    }

    public int getCapaciteMax() {
        return capaciteMax;
    }

    public void setCapaciteMax(int capaciteMax) {
        this.capaciteMax = capaciteMax;
    }

    private static int generateId() {
        return nextId++; // Incrémente le prochain identifiant et retourne l'ancienne valeur
    }

    @Override
    public String toString() {
        return "Galeries{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", adresse='" + adresse + '\'' +
                ", ville='" + ville + '\'' +
                ", pays='" + pays + '\'' +
                ", capaciteMax=" + capaciteMax +
                '}';
    }
}

File: src/Entity/Galeries.php
<?php

namespace App\Entity;

use App\Repository\GaleriesRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: GaleriesRepository::class)]
class Galeries
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private $id;

    #[ORM\Column(length:255)]
    #[Assert\NotBlank(message: "Name is required !")]
    private $nom = null;

    #[ORM\Column(length:255)]
    #[Assert\NotBlank(message: "Address is required !")]
    private $adresse = null;

    #[ORM\Column(length:255)]
    #[Assert\NotBlank(message: "City is required !")]
    private $ville = null;

    #[ORM\Column(length:255)]
    #[Assert\NotBlank(message: "Country is required !")]
    private $pays = null;

    #[ORM\Column]
    #[Assert\Positive(message: "Capacity must be greater then 0")]
    private $capaciteMax = NULL;

    #[ORM\Column(length:255)]
    private $image;

    public function getId(): ?string
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): static
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getVille(): ?string
    {
        return $this->ville;
    }

    public function setVille(string $ville): static
    {
        $this->ville = $ville;

        return $this;
    }

    public function getPays(): ?string
    {
        return $this->pays;
    }

    public function setPays(string $pays): static
    {
        $this->pays = $pays;

        return $this;
    }

    public function getCapaciteMax(): ?string
    {
        return $this->capaciteMax;
    }

    public function setCapaciteMax(string $capaciteMax): static
    {
        $this->capaciteMax = $capaciteMax;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): static
    {
        $this->image = $image;

        return $this;
    }

}